File: packages/ts/test/config/networks.test.ts
import { describe, it, expect } from "vitest";
import { getNetwork, NETWORKS } from "../../src/config/networks";

describe("networks", () => {
  it("testnet is fully populated and live", () => {
    const n = getNetwork("testnet");
    expect(n.live).toBe(true);
    expect(n.chainId).toBe("qorechain-diana");
    expect(n.coin).toEqual({ display: "QOR", base: "uqor", exponent: 6 });
    expect(n.bech32.account).toBe("qor");
    expect(n.bech32.validator).toBe("qorvaloper");
    expect(n.endpoints?.rest).toContain("1317");
    expect(n.endpoints?.evmRpc).toContain("8545");
    expect(n.endpoints?.svmRpc).toContain("8899");
  });

  it("mainnet is fully populated and live", () => {
    expect(NETWORKS.mainnet.live).toBe(true);
    expect(NETWORKS.mainnet.chainId).toBe("qorechain-vladi");
    expect(NETWORKS.mainnet.bech32.account).toBe("qor");
    expect(NETWORKS.mainnet.bech32.validator).toBe("qorvaloper");
    expect(NETWORKS.mainnet.bech32.consensus).toBe("qorvalcons");
    expect(NETWORKS.mainnet.coin).toEqual({
      display: "QOR",
      base: "uqor",
      exponent: 6,
    });
    expect(NETWORKS.mainnet.endpoints?.rest).toContain("1317");
    expect(NETWORKS.mainnet.endpoints?.evmRpc).toContain("8545");
    expect(NETWORKS.mainnet.endpoints?.svmRpc).toContain("8899");
  });

  it("getNetwork returns the live mainnet config", () => {
    const n = getNetwork("mainnet");
    expect(n.live).toBe(true);
    expect(n.chainId).toBe("qorechain-vladi");
  });
});

File: packages/go/qorechain/networks/networks_test.go
package networks

import (
	"strings"
	"testing"
)

func TestGetNetworkTestnet(t *testing.T) {
	cfg, err := GetNetwork("testnet")
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	if cfg.ChainID != "qorechain-diana" {
		t.Errorf("chain id = %q, want qorechain-diana", cfg.ChainID)
	}
	if !cfg.Live {
		t.Error("testnet should be live")
	}
	if cfg.Endpoints == nil {
		t.Fatal("testnet endpoints should not be nil")
	}
	if cfg.Endpoints.REST != "http://localhost:1317" {
		t.Errorf("rest = %q", cfg.Endpoints.REST)
	}
	if cfg.Endpoints.GRPC != "http://localhost:9090" {
		t.Errorf("grpc = %q", cfg.Endpoints.GRPC)
	}
	if cfg.Endpoints.RPC != "http://localhost:26657" {
		t.Errorf("rpc = %q", cfg.Endpoints.RPC)
	}
	if cfg.Endpoints.EVMRPC != "http://localhost:8545" {
		t.Errorf("evmrpc = %q", cfg.Endpoints.EVMRPC)
	}
	if cfg.Endpoints.EVMWS != "ws://localhost:8546" {
		t.Errorf("evmws = %q", cfg.Endpoints.EVMWS)
	}
	if cfg.Endpoints.SVMRPC != "http://localhost:8899" {
		t.Errorf("svmrpc = %q", cfg.Endpoints.SVMRPC)
	}
	if cfg.Bech32.Account != "qor" || cfg.Bech32.Validator != "qorvaloper" {
		t.Errorf("bech32 prefixes wrong: %+v", cfg.Bech32)
	}
	if cfg.Coin.Display != "QOR" || cfg.Coin.Base != "uqor" || cfg.Coin.Exponent != 6 {
		t.Errorf("coin info wrong: %+v", cfg.Coin)
	}
}

func TestGetNetworkMainnetLive(t *testing.T) {
	cfg, err := GetNetwork("mainnet")
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	if !cfg.Live {
		t.Error("mainnet should be live")
	}
	if cfg.ChainID != "qorechain-vladi" {
		t.Errorf("chain id = %q, want qorechain-vladi", cfg.ChainID)
	}
	if cfg.Endpoints == nil {
		t.Fatal("mainnet endpoints should not be nil")
	}
	if cfg.Endpoints.REST != "http://localhost:1317" {
		t.Errorf("rest = %q", cfg.Endpoints.REST)
	}
	if cfg.Bech32.Account != "qor" || cfg.Bech32.Validator != "qorvaloper" || cfg.Bech32.Consensus != "qorvalcons" {
		t.Errorf("bech32 prefixes wrong: %+v", cfg.Bech32)
	}
	if cfg.Coin.Display != "QOR" || cfg.Coin.Base != "uqor" || cfg.Coin.Exponent != 6 {
		t.Errorf("coin info wrong: %+v", cfg.Coin)
	}
}

func TestGetNetworkUnknown(t *testing.T) {
	_, err := GetNetwork("nope")
	if err == nil {
		t.Fatal("expected error for unknown network")
	}
	if !strings.Contains(err.Error(), "unknown network") {
		t.Errorf("error = %q", err.Error())
	}
}

func TestMainnetPresetShape(t *testing.T) {
	cfg := Networks["mainnet"]
	if !cfg.Live {
		t.Error("mainnet should be live")
	}
	if cfg.ChainID != "qorechain-vladi" {
		t.Errorf("mainnet chain id should be qorechain-vladi, got %q", cfg.ChainID)
	}
	if cfg.Endpoints == nil {
		t.Error("mainnet endpoints should not be nil")
	}
}
